sources.sh

```diff
@@ -219,6 +219,22 @@ jq <<<"$sources" --argjson pins "$externalPinsJson" '
 		# https://unix.stackexchange.com/a/738744/153467
 		reduce .[] as $a ([]; if IN(.[]; $a) then . else . += [$a] end)
 	;
+	def meld($a; $b):
+		# recursive merge of objects like "*", but also append lists (uniquely) instead of replace
+		# https://stackoverflow.com/a/53666584, but with lists unique-ified
+		if ($a | type) == "object" and ($b | type) == "object" then
+			reduce ([$a, $b] | add | keys_unsorted[]) as $k ({};
+				.[$k] = meld( $a[$k]; $b[$k])
+			)
```

Comment:
This is doing a lot of work twice :joy:

```suggestion
			reduce ([ $a, $b | keys_unsorted[] ] | unique_unsorted[]) as $k ({};
				.[$k] = meld($a[$k]; $b[$k])
			)
```